<launch>
    <arg name="map_size_x_up" default="10.0"/>
    <arg name="map_size_y_up" default="10.0"/>
    <arg name="map_size_x_lo" default="-10.0"/>
    <arg name="map_size_y_lo" default="-10.0"/>

    <arg name="start_x" default=" 0.0"/>
    <arg name="start_y" default=" 0.0"/>
    <arg name="map_resolution_" default=" 0.05"/>
    <arg name="cube_iter" default=" 50"/>
    <arg name="expand_step" default=" 1"/>

    <arg name="_is_limit_vel" default=" true"/>
    <arg name="_is_limit_acc" default=" true"/>

    <node pkg="navigation" type="btraj_node" name="btraj_node" output = "screen"  launch-prefix="xterm -e gdb -ex run --args">
        <param name="map/resolution"   value="$(arg map_resolution_)" />
        <param name="map/x_up_bound"       value="$(arg map_size_x_up)"/>
        <param name="map/y_up_bound"       value="$(arg map_size_y_up)"/>
        <param name="map/x_low_bound"       value="$(arg map_size_x_lo)"/>
        <param name="map/y_low_bound"       value="$(arg map_size_y_lo)"/>
        <param name="map/x_map_index" value="400"/>
        <param name="map/y_map_index" value="400"/>
        <param name="planning/start_x" value="$(arg start_x)"/>
        <param name="planning/start_y" value="$(arg start_y)"/>
        <param name="planning/iteration_num" value="$(arg cube_iter)"/>
        <param name="planning/expand_step" value="$(arg expand_step)"/>
        <param name="planning/max_acc" value="1.0"/>
        <param name="planning/max_vel" value="1.0"/>
        <param name="planning/time_brk" value="1.05"/>
        <param name="planning/is_limit_vel" value="$(arg  _is_limit_vel)"/>
        <param name="planning/is_limit_acc" value="$(arg  _is_limit_acc)"/>
        <param name="planning/target_frame" value = "map"/>
        <param name="planning/traj_order" value="5"/>
        <param name="planning/min_order" value="2.5"/>
        <param name="planning/order_min" value="3"/>
        <param name="planning/order_max" value="12"/>
        <param name="planning/margin" value="0.05"/>
        <param name="planning/replan_thresh" value="2.0"/>
        <param name="planning/no_replan_thresh" value="2.0"/>
        <param name="planning/arrive_thresh" value="0.1"/>
        <remap from="map" to="correct_pointcloud"/>
        <remap from="odom" to="RosAria/pose"/>
    </node>

    <node pkg="navigation" type="perception_node" name="perception_node" output="log">
        <remap from="odom" to="RosAria/pose"/>
        <rosparam file="$(find navigation)/config/lidar_undistortion.yaml" command="load"/>
    </node>

    <node pkg="hokuyo_node" type="hokuyo_node" name="hokuyo_node">
        <param name="port" type="string" value="/dev/ttyACM0"/>
    </node>

    <node pkg="tf" type="static_transform_publisher" name="base_to_laser" args="0.1 0.0 0.05 0.0 0.0 0.0 base_link laser 100"/>

    <include file="$(find controller)/launch/controller.launch"/>
    <include file="$(find gmapping)/launch/slam_test.launch"/>
    <include file="$(find rosaria)/launch/rosaria.launch"/>
     <node name="rviz" pkg="rviz" type="rviz" output = "screen" args="-d $(find navigation)/launch/xtark_mapping.rviz" />
  
</launch>